parse.go

```diff
@@ -73,37 +73,64 @@ func Parse(v string) (GppContainer, error) {
 		}
 	}
 	if len(secIDs) != secCount {
-		return gpp, fmt.Errorf("error parsing GPP header, section IDs do not match the number of sections: found %d IDs, have %d sections", len(secIDs), secCount)
+		return gpp, []error{fmt.Errorf("error parsing GPP header, section IDs do not match the number of sections: found %d IDs, have %d sections", len(secIDs), secCount)}
 	}
 	gpp.SectionTypes = secIDs
 
 	sections := make([]Section, secCount)
+	var errs []error
 	for i, id := range secIDs {
 		switch id {
 		case constants.SectionUSPNAT:
 			sections[i], err = uspnat.NewUSPNAT(sectionStrings[i+1])
+			if err != nil {
+				errs = append(errs, sectionParseError(int(id), err))
+			}
 		case constants.SectionUSPCA:
 			sections[i], err = uspca.NewUSPCA(sectionStrings[i+1])
+			if err != nil {
+				errs = append(errs, sectionParseError(int(id), err))
+			}
 		case constants.SectionUSPVA:
 			sections[i], err = uspva.NewUSPVA(sectionStrings[i+1])
+			if err != nil {
+				errs = append(errs, sectionParseError(int(id), err))
+			}
 		case constants.SectionUSPCO:
 			sections[i], err = uspco.NewUSPCO(sectionStrings[i+1])
+			if err != nil {
+				errs = append(errs, sectionParseError(int(id), err))
+			}
 		case constants.SectionUSPUT:
 			sections[i], err = usput.NewUSPUT(sectionStrings[i+1])
+			if err != nil {
+				errs = append(errs, sectionParseError(int(id), err))
+			}
 		case constants.SectionUSPCT:
 			sections[i], err = uspct.NewUSPCT(sectionStrings[i+1])
+			if err != nil {
+				errs = append(errs, sectionParseError(int(id), err))
+			}
 		default:
 			sections[i] = GenericSection{sectionID: id, value: sectionStrings[i+1]}
+			if err != nil {
+				errs = append(errs, sectionParseError(int(id), err))
+			}
 		}
 	}
 
-	if err != nil {
-		return gpp, err
-	}
-
 	gpp.Sections = sections
 
-	return gpp, nil
+	return gpp, errs
+}
+
+func sectionParseError(id int, err error) error {
+	if id < len(constants.SectionIDNames) && id > 0 {
+		return fmt.Errorf("Error parsing %s consent string: %s", constants.SectionIDNames[id], err)
+	} else {
+		return fmt.Errorf("Error parsing UNKNOWN consent string: %s", err)
```

Comment:
Perhaps it's better to say unsupported for future compatibility and include the section id? Something like:
```
fmt.Errorf("error parsing unsupported (section %d) consent string: %s", err)
```

Reply:
Sounds good. I think I was overthinking things with this helper function anyway. :)

Reply:
:)